lib/terrafying/components/vpn_oidc.rb

```diff
@@ -0,0 +1,315 @@
+            ports: [22, 443, { number: 1194, type: 'udp' }],
+            tags: options[:tags],
+            units: units + options[:units],
+            files: files,
+            keypairs: keypairs,
+            subnets: options[:subnets],
+            instances: instances,
+            iam_policy_statements: [
+              {
+                Effect: 'Allow',
+                Action: [
+                  'ec2:DescribeRouteTables'
+                ],
+                Resource: [
+                  '*'
+                ]
+              }
+            ]
+          }.merge(options[:service])
+        )
+
+        @ip_address = @service.instance_set.instances[0].ip_address
+
+        self
+      end
+
+      def allow_security_group_in(vpc, name: '')
+        name = "allow-#{@vpc.name}-vpn".downcase if name.empty?
+
+        ingress_rules = [
+          {
+            from_port: 0,
+            to_port: 0,
+            protocol: -1,
+            security_groups: [@service.egress_security_group]
+          }
+        ]
+
+        if @is_public
+          ingress_rules << {
+            from_port: 0,
+            to_port: 0,
+            protocol: -1,
+            cidr_blocks: ["#{@ip_address}/32"]
+          }
+        end
+
+        resource :aws_security_group, tf_safe("#{name}-#{vpc.name}"),
+                 name: name,
+                 vpc_id: vpc.id,
+                 ingress: ingress_rules
+      end
+
+      def openvpn_service
+        Ignition.container_unit(
+          'openvpn', 'kylemanna/openvpn',
+          host_networking: true,
+          privileged: true,
+          volumes: [
+            '/etc/ssl/openvpn:/etc/ssl/openvpn:ro',
+            '/etc/openvpn:/etc/openvpn'
+          ],
+          required_units: ['docker.service', 'network-online.target', 'openvpn-authz.service']
+        )
+      end
+
+      def openvpn_authz_service(route_all_traffic, route_dns_entry, groups, oidc)
+        optional_arguments = []
+
+        optional_arguments << '--route-all' if route_all_traffic
+
+        optional_arguments += groups.map { |group| "--oidc-allowed-groups \"#{group}\"" }
+        optional_arguments += route_dns_entry.map { |entry| "--route-dns-entries #{entry}" }
+
+        Ignition.container_unit(
+          'openvpn-authz', 'quay.io/uswitch/openvpn-authz:2.0',
+          host_networking: true,
+          volumes: [
+            '/etc/ssl/openvpn:/etc/ssl/openvpn',
+            '/var/openvpn-authz:/var/openvpn-authz'
+          ],
+          environment_variables: [
+            "AWS_REGION=#{aws.region}"
+          ],
+          arguments: optional_arguments + [
+            "--http-address http://127.0.0.1:8080",
+            "--fqdn #{@fqdn}",
+            '--cache /var/openvpn-authz',
+            "--oidc-client-id \"#{oidc[:client_id]}\"",
+            "--oidc-issuer-url \"#{oidc[:issuer_url]}\"",
+            '/etc/ssl/openvpn'
+          ]
+        )
+      end
+
+      def caddy_service(ca)
```

Comment:
Given that all this does it terminate TLS now can we just move this into openvpn-authz?

Reply:
Can do